server/src/index.js

```diff
@@ -26,31 +26,31 @@ const main = async () => {
     if (data.type === 'news') {
```

Comment:
Also let's delay creating the new object which is to be inserted into the database until before insertion, since different types of posts have different fields.